core/src/main/java/com/es/core/service/cart/HttpSessionCartService.java

```diff
@@ -1,29 +1,62 @@
 package com.es.core.service.cart;
 
+import com.es.core.dao.phone.PhoneDao;
 import com.es.core.model.cart.Cart;
+import com.es.core.model.cartItem.CartItem;
+import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.stereotype.Service;
 
+import java.math.BigDecimal;
+import java.util.ArrayList;
+import java.util.List;
 import java.util.Map;
+import java.util.Optional;
+import java.util.stream.Collectors;
 
 @Service
 public class HttpSessionCartService implements CartService {
-    @Override
-    public Cart getCart() {
-        throw new UnsupportedOperationException("TODO");
-    }
+
+    @Autowired
+    private Cart cart;
+
+    @Autowired
+    PhoneDao phoneDao;
 
     @Override
     public void addPhone(Long phoneId, Long quantity) {
-        throw new UnsupportedOperationException("TODO");
+        cart.addItem(phoneId, quantity);
     }
 
     @Override
     public void update(Map<Long, Long> items) {
-        throw new UnsupportedOperationException("TODO");
+        throw new UnsupportedOperationException();
     }
 
     @Override
     public void remove(Long phoneId) {
-        throw new UnsupportedOperationException("TODO");
+        cart.removeItem(phoneId);
+    }
+
+    @Override
+    public Long countTotalItem() {
+        return cart.getItems().values().stream().mapToLong(i -> i).sum();
+    }
+
+    @Override
+    public BigDecimal countTotalSum() {
```

Comment:
Let implement it this way. Create request-scoped `CartView` component. Add all the properties you need (like subtotal and number of items) to that bean. Create `@ControllerAdvice`, add cart view to the model using `@ModelAttribute`